src/csharp/Grpc.Core/GrpcEnvironment.cs

```diff
@@ -311,13 +328,16 @@ internal static void GrpcNativeShutdown()
         /// </summary>
         private async Task ShutdownAsync()
         {
-            if (isClosed)
+            if (isShutdown)
```

Comment:
It's fine as ShutdownAsync() is private and only called from GrpcEnvironment.ReleaseAsync().